src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+/**
+ * Represents a service locator able to lazy load a given range of services.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @experimental in version 3.3
+ */
+class ServiceLocatorArgument implements ArgumentInterface
+{
+    private $values;
+
+    /**
+     * @param array $values An array of references entries indexed by identifier
```

Comment:
`Reference[]` + validity check in the constructor? maybe also in loaders if that can provide more useful error messages?